src/deweb/immutable.ts

```diff
@@ -0,0 +1,32 @@
+import { Provider } from 'src/provider'
+import { Operation } from 'src/operation'
+import { PublicAPI } from '../client'
+
+export async function makeImmutable(
+  address: string,
+  provider: Provider,
+  waitFinal = false
+): Promise<Operation> {
+  const operation = await provider.callSC({
+    func: 'upgradeSC',
+    target: address,
+  })
+
+  await (waitFinal
+    ? operation.waitFinalExecution()
+    : operation.waitSpeculativeExecution())
+  return operation
+}
+
+export async function isImmutable(
+  address: string,
+  nodeUrl: string,
```

Comment:
its strange to provide an url here.
It would best to use the provider in function parameter, and get the url from it ( using networkInfos). you can also handle the case where nodeUrl is not available (its the case for bearby) and use the public one based on the chain ID